src/odemis/acq/align/__init__.py

```diff
@@ -153,35 +147,42 @@ def _cancelTurnLightAndCheck(f):
             logging.debug("The task already finished")
             return False
         f._task_state = CANCELLED
-        f._done.set()
         logging.debug("Task cancelled.")
     return True
 
 
-def _doTurnLightAndCheck(f, bl, ccd):
-    f._done.clear()
+def _doTurnOnLightAndCheck(f, bl, ccd):
     try:
         # check if the light is already turned on. The light should normally be turned off.
         # In case it's already turned on return and start the procedure of autofocus
         if bl.emissions.value[0] * bl.power.value != 0:
+            logging.debug("The light is already on")
             return
         if f._task_state == CANCELLED:
             raise CancelledError()
 
         # Light turned off, take the avg intensity of the image
-        img_light_off = ccd.data.get()
+        img_light_off = ccd.data.get(asap=False)
         avg_intensity_off = numpy.average(img_light_off)
         # set the avg minimum intensity - a significant change (+50%)
         avg_intensity_min_on = avg_intensity_off * 1.5 + 0.1
         # Turn the light on
         bl.power.value = bl.power.range[1]
         bl.emissions.value = [1] * len(bl.emissions.value)
-        img2 = ccd.data.get()
-        # Check whether 0.5 percent of the total pixels have intensity more than the avg minimum
-        while (numpy.sum(img2 > avg_intensity_min_on))/(img2.shape[0]*img2.shape[1]) < 0.005:
+        while True:
             img2 = ccd.data.get()
-        logging.debug("the light is on")
-        return
+            # number of pixels with higher intensity than the avg minimum
+            pixels_high_intensity = numpy.sum(img2 > avg_intensity_min_on)
+            img_size = img2.shape[0]*img2.shape[1]
+            # the percent of pixels that have intensity higher than the avg minimum
+            a = pixels_high_intensity/img_size
+            if f._task_state == CANCELLED:
+                raise CancelledError()
+            # check whether this percent is larger than 0.5% which indicates that the light is on
+            if (a > 0.005):
+                logging.debug("The light is on")
```

Comment:
In the logging, please indicate the percentage of pixel with "high intensity", and also put a log statement in case it didn't pass, with that percentage. This way, in case a system has an issue and the light is never detected as turned on, it'll be a little easier to debug.